src/zh/zaimanhua/src/eu/kanade/tachiyomi/extension/zh/zaimanhua/ZaimanhuaDto.kt

```diff
@@ -0,0 +1,138 @@
+    private val chapterGroups: List<ChapterGroupDto>,
+) {
+    fun toSManga() = SManga.create().apply {
+        url = id.toString()
+        title = this@MangaDto.title
+        author = authors.joinToString { it.name }
+        description = this@MangaDto.description
+        genre = types.joinToString { it.name }
+        status = parseStatus(this@MangaDto.status[0].name)
+        thumbnail_url = cover
+        initialized = true
+    }
+
+    fun parseChapterList(): List<SChapter> {
+        val mangaId = id.toString()
+        val size = chapterGroups.sumOf { it.size }
+        return chapterGroups.flatMapTo(ArrayList(size)) {
+            it.toSChapterList(mangaId)
+        }
+    }
+}
+
+@Serializable
+class ChapterGroupDto(
+    private val title: String,
+    private val data: List<ChapterDto>,
+) {
+    fun toSChapterList(mangaId: String): List<SChapter> {
+        val groupName = title
+        val isDefaultGroup = groupName == "连载"
+        return data.map {
+            it.toSChapterInternal().apply {
+                url = "$mangaId/$url"
+                if (!isDefaultGroup) name = "$groupName: $name"
+            }
+        }
+    }
+
+    val size get() = data.size
+}
+
+@Serializable
+class ChapterDto(
+    @SerialName("chapter_id")
+    private val id: Int,
+    @SerialName("chapter_title")
+    private val name: String,
+    @SerialName("updatetime")
+    private val updateTime: Long = 0,
+) {
+    fun toSChapterInternal() = SChapter.create().apply {
+        url = id.toString()
+        name = this@ChapterDto.name.formatChapterName()
+        date_upload = updateTime * 1000
+    }
+}
+
+@Serializable
+class ChapterImagesDto(
+    @SerialName("page_url")
+    val lowResImages: List<String>,
+    @SerialName("page_url_hd")
+    val images: List<String>,
+)
+
+@Serializable
+class PageDto(
+    val list: List<PageItemDto>,
+    val page: Int,
+    val size: Int,
+    val total: Int,
+) {
+    fun toMangasPage(): MangasPage {
+        val hasNextPage = page * size < total
+        return MangasPage(list.map { it.toSManga() }, hasNextPage)
+    }
+}
+
+@Serializable
+class PageItemDto(
+    @JsonNames("comic_id")
+    private val id: Int,
+    private val title: String,
+    private val authors: String = "",
+    private val status: String,
+    private val cover: String,
+    private val types: String,
+) {
+    fun toSManga() = SManga.create().apply {
+        url = this@PageItemDto.id.toString()
+        title = this@PageItemDto.title
+        author = authors.formatList()
+        genre = types.formatList()
+        status = parseStatus(this@PageItemDto.status)
+        thumbnail_url = cover
+    }
+}
+
+@Serializable
+class TagDto(
+    @SerialName("tag_name")
+    val name: String,
+)
+
+@Serializable
+class UserDto(
+    val user: UserInfoDto,
+) {
+    @Serializable
+    class UserInfoDto(
+        val token: String,
+    )
+}
+
+@Serializable
+class ResponseDto<T>(
+    val errno: Int = 0,
+    val data: T?,
```

Comment:
why make it nullable when you are using `!!` non-null assertions in the rest of code